### ISSUE TITLE
Deprecated HiveFactory Support

### ISSUE THREAD

**Issue by serefacet** [confluentinc/kafka-connect-storage-common#179]
Hello all,

I'm just curious why the hiveFactory method is [deprecated](https://github.com/confluentinc/kafka-connect-storage-common/blob/master/format/src/main/java/io/confluent/connect/storage/format/Format.java#L29) in Format class. I'm not able to find any explanation as a replacement and also it's used by [kafka-connect-hdfs](https://github.com/confluentinc/kafka-connect-hdfs) connector as well.
 
I'm asking this question because we are thinking about adding Hive partitions from Kafka S3 Connector in order to access the data whenever it's transferred to S3. Now, we are handling this requirement with a scheduled workflow that checks the partition folders in S3 then adds corresponding Hive partitions if not exists. But, we would like to handle that through Kafka Connect. 

To sum up, we just want to check the project roadmap before implementing this feature. If the hive support is no longer considered in the future, we don't want to locked into the specific Kafka Connect versions.

Thank you in advance

**Comment by raphaelauv** [comment]
Look like hive support is not about to get remove from storage-commun -> https://github.com/confluentinc/kafka-connect-storage-common/pull/203
